src/features/Character/queries/trait.ts

```diff
@@ -0,0 +1,45 @@
+import { QueryObserverOptions, useQuery, UseQueryResult } from 'react-query';
+import { defaultFetchStrategy, defaultStaleTime } from './config';
+import { TraitsService } from '../../../api/services/traitsService';
+import { CharacterTrait } from '../../../models/characters/characterTrait';
+import { TraitsWithRoots } from '../../../models/traitsWithRoots';
+import { SpoilerLevel } from '../../../models/spoilerLevel';
+import { ExtendedTraitsWithRoots } from '../../../models/extendedTraitWithRoots';
+
+/**
+ * Extends root traits with spoiler level.
+ * @param characterTraits List of character traits.
+ */
+const extendTraits = async(
+  characterTraits: readonly CharacterTrait[],
+  traitsWithRoot: Promise<TraitsWithRoots>,
+): Promise<ExtendedTraitsWithRoots> => {
+  const { traits, rootTraits } = await traitsWithRoot;
+  const extendedTraits = traits.map(
+    trait => ({
+      ...trait,
+      spoilerLevel: characterTraits.find(ct => ct.id === trait.id)?.spoilerLevel ?? SpoilerLevel.None,
```

Comment:
I think default spoiler level should be highest. 
What if death of character mistakenly have no `spoilerLevel` property? 🫤